clients/deltalake/build.gradle.kts

```diff
@@ -75,5 +75,8 @@ nessieQuarkusApp {
 forceJava11ForTests()
 
 tasks.named<ShadowJar>("shadowJar") {
-  dependencies { include(dependency("org.projectnessie:.*:.*")) }
+  dependencies {
+    include(dependency("org.projectnessie.nessie:.*:.*"))
+    include(dependency("org.projectnessie.nessie-integrations:.*:.*"))
```

Comment:
This will include Iceberg stuff, I guess. Would it make sense to limit this to deltalake? I know it matches old behaviour, but since we're repackaging artifacts, I guess it might be worth isolating unrelated artifacts? WDYT?

Reply:
Idea is exactly to move everything, especially the Iceberg-related stuff, to the `org.projectnessie.nessie-integrations` group-ID

Reply:
Is Iceberg stuff relevant to the deltalake client? Why do we bundle it into the client jar?

Reply:
It's an integration as well

Reply:
Added a note of this to #4066 for follow-up.